SilverSolver/assignment1/Q2.py

```diff
@@ -0,0 +1,38 @@
+class single_linked_node:
+    def __init__(self, value=None, child=None):
+        self.child = child
+        self.value = value
+
+def return_k_th_from_end_element(k, start_node):
+    list_len = 0
+    current_node = start_node
+    while (current_node):
+        list_len += 1
+        current_node = current_node.child
+    element_num = list_len - k
+    if element_num < 0 or element_num > list_len:
+        print("wrong k")
```

Comment:
It is more useful and more common to raise Exception in case of problems.

For example:
>>> [1,2,3][10]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range

Do you think you can raise an Index error instead of the print side effect?